tests/ytt/external_db_test.go

```diff
@@ -0,0 +1,69 @@
+package ytt
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("External DB", func() {
+
+	var ctx RenderingContext
+	var data map[string]string
+	var templates []string
+
+	BeforeEach(func() {
+		templates = []string{
+			pathToFile("config/postgres"),
+			pathToFile("tests/ytt/postgres/postgres-values.yml"),
+		}
+	})
+
+	JustBeforeEach(func() {
+		ctx = NewRenderingContext(templates...).WithData(data)
+	})
+
+	Context("disabled", func() {
+
+		BeforeEach(func() {
+			data = map[string]string{
+				"cf_db.admin_password":   "vz9o2hbkh6x7ztnd6ua6",
+				"capi.database.password": "9v5ljlmxje8uv32gwl5q",
+				"uaa.database.password":  "yevxad9e0pvgc8l6osnt",
+				"capi.database.host":     "",
+				"uaa.database.host":      "",
+			}
+		})
+
+		It("should have cf-db namespace and cf-db-postgresql statefulset", func() {
+
+			Expect(ctx).To(ProduceYAML(
+				And(
+					WithNamespace("cf-db"),
+					WithStatefulSet("cf-db-postgresql").WithNamespace("cf-db")),
```

Comment:
Is there ever a case where you would want to assert on a StatefulSet and not assert on its namespace? I understand the composability of Gomega matchers, but I feel that it complicates the readability of the test.

(This comment makes more sense in the context of my comment about "negative" tests below.)

Reply:
Maybe not?  In general it seems things are always put into a namespace even if it is "default".  So perhaps it makes  sense to expand the signature to `WithStatefulSet(name, namespace string)` and just be explicit about it.

Reply:
Could we do a matcher that nests inside of ProduceYaml to specify namespace before any other matchers? Will we ever be asserting across namespaces?